### ISSUE TITLE
Define standard PolicyAncestor status type for Policy objects

### ISSUE THREAD

**Issue by youngnick** [kubernetes-sigs/gateway-api#2923]
Spun out of #2813 (Specifically, https://github.com/kubernetes-sigs/gateway-api/pull/2813#discussion_r1520568442).

**What would you like to be added**:

We need either an update to GEP-713, GEP-2648, and GEP-2649 or to make a small GEP under this issue to properly discuss the idea of a PolicyAncestorStatus field, as in BackendTLSPolicy.

The PolicyAncestorStatus field is a status field, based on the ParentStatus in Route objects, that describes its status relative to the named ancestor object. "Ancestor" is used instead of "Parent" (as in Route) because for Policy objects, the relevant object for status is not always going to be the `targetRef` of the Policy.

**Why this is needed**:
To further ease the use of the Policy Attachment API pattern.

**Comment by mikemorris** [comment]
For additional context, an initial implementation of this pattern was first introduced in https://github.com/kubernetes-sigs/gateway-api/pull/2448

**Comment by mikemorris** [comment]
Finally taking some time to dig deeper into this, and my initial thought is that at a minimum the name for this type feels incorrect, because it only feels applicable to policies which are targeted at a lower-hierarchy resource which have implications on the behavior of a higher-hierarchy resource, and would not be appropriate for modelling an inverse relationship or peer relationships (such as an AuthorizationPolicy between a Pod/workload and a Service or other Pod/workload).

It also feels like there's tension between this approach and the suggestion for Inherited Policies in https://github.com/kubernetes-sigs/gateway-api/pull/2813 for a controller to directly modify the conditions of affected resources by setting a *PolicyAffected condition or label _on the affected resource_ rather than centralized in the Policy status.

**Comment by mikemorris** [comment]
/remove-lifecycle rotten

This is probably duplicative of #738 now, but not sure how to best consolidate.

**Comment by youngnick** [comment]
Agreed, let's definitely keep this open to come back to in the 1.13 timeframe.

**Comment by youngnick** [comment]
/remove-lifecycle rotten

Probably should be addressed after #3609 merges.
